packages/mcp-cloudflare/src/server/routes/rag-search.ts

```diff
@@ -0,0 +1,181 @@
+import { Hono } from "hono";
+import type { Env } from "../types";
+import { logError } from "@sentry/mcp-server/logging";
+import { z } from "zod";
+import type {
+  AutoRAGSearchResponse,
+  AutoRAGSearchResult,
+} from "./rag-search-types";
+import type { RateLimitResult } from "../types/chat";
+
+// Request schema matching the MCP tool parameters
+const SearchRequestSchema = z.object({
+  query: z.string().trim().min(1, "Query is required"),
+  maxResults: z.number().int().min(1).max(10).default(3).optional(),
+  docType: z
+    .enum(["all", "api", "sdk", "platform", "product"])
```

Comment:
this will likely end up being like "sdk", "api", and "contributor" or something similar.. tbd here